From b8f068e7b9a6f1e8c0cb59dac47cc0f6f54c97f0 Mon Sep 17 00:00:00 2001
From: Leonardo Taccari <iamleot@gmail.com>
Date: Thu, 31 Oct 2024 13:14:08 +0100
Subject: [PATCH] Force usage of pytz

For timezone functionalities icalendar can use both pytz and zoneinfo,
preferring zoneinfo.

We rely on pytz for several attributes, stick to it.

Fixes #1361.
Should obsolete #1363.
---
 CHANGELOG.rst     | 1 +
 khal/icalendar.py | 4 ++++
 pyproject.toml    | 2 +-
 3 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/CHANGELOG.rst b/CHANGELOG.rst
index 47cb66302..ca48b4d55 100644
diff --git a/khal/icalendar.py b/khal/icalendar.py
index 39e6eda8f..3d1fcd71f 100644
--- a/khal/icalendar.py
+++ b/khal/icalendar.py
@@ -37,6 +37,10 @@
 
 logger = logging.getLogger('khal')
 
+# Force use of pytz because we rely on functionalities not available in
+# zoneinfo.
+icalendar.use_pytz()
+
 
 def split_ics(ics: str, random_uid: bool=False, default_timezone=None) -> List:
     """split an ics string into several according to VEVENT's UIDs
diff --git a/pyproject.toml b/pyproject.toml
index 597a6046a..7dc930926 100644
